include/SDL3/SDL_intrin.h

```diff
@@ -108,27 +113,39 @@ _m_prefetch(void *__P)
 #endif
 
 #if defined(__loongarch_sx) && !defined(SDL_DISABLE_LSX)
-#include <lsxintrin.h>
+# define SDL_LSX_INTRINSICS 1
+# include <lsxintrin.h>
 #endif
+
 #if defined(__loongarch_asx) && !defined(SDL_DISABLE_LASX)
-#include <lasxintrin.h>
+# define SDL_LASX_INTRINSICS 1
+# include <lasxintrin.h>
 #endif
+
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
-#if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
-#include <immintrin.h>
-#endif
-#if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
-#include <mmintrin.h>
-#endif
-#if (defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
-#include <xmmintrin.h>
-#endif
-#if (defined(__SSE2__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE2)
-#include <emmintrin.h>
-#endif
-#if (defined(__SSE3__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE3)
-#include <pmmintrin.h>
-#endif
-#endif /**/
+# if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
+#  define SDL_AVX_INTRINSICS
+#  include <immintrin.h>
+# endif
+# if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
+#  define SDL_MMX_INTRINSICS 1
+#  include <mmintrin.h>
+# endif
+# if ((defined(_MSC_VER) && defined(_M_IX86) && _M_IX86_FP == 1) || defined(_M_X64) || defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
+   /* x86 MSVC defines _M_IX86_FP == 1 when compiled with /arch:SSE */
+#  define SDL_SSE_INTRINSICS 1
+#  include <xmmintrin.h>
+# endif
+# if ((defined(_MSC_VER) && defined(_M_IX86) && _M_IX86_FP >= 1) || defined(_M_X64) || defined(__SSE2__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE2)
+   /* x86 MSVC defines _M_IX86_FP == 2 when compiled with /arch:SSE2 */
+#  define SDL_SSE2_INTRINSICS 1
+#  include <emmintrin.h>
+# endif
+# if (defined(_M_X64) || defined(__SSE3__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE3)
+   /* x86 MSVC does not provide macro's to detect SSE3/SSE4 */
```

Comment:
Why do we not enable SSE3 with MSVC for `_M_X86` too ?

Reply:
I googled around, but found no reference to MSVC  macro's advertising SSE3 support. So I left it at that.

Reply:
Possibly the same situation with SSE[2], i.e. possibly no condition needed